common/pipeline_monitor/README.md

```diff
@@ -0,0 +1,216 @@
+
+1. Add **masterreader** role to service principal which is same name as the Function App.
+
+### Install Tools
+
+1. Install [Azure CLI version 2.4 or later](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli).
+
+1. Install [Azure Functions Core Tools version 3.x](https://docs.microsoft.com/en-us/azure/azure-functions/functions-run-local?tabs=windows%2Ccsharp%2Cbash#v2).
+
+1. Python 3.8 (64-bit), Python 3.7 (64-bit), Python 3.6 (64-bit), which are all supported by version 3.x of Azure Functions. As this sample uses Python 3.8, make sure by running `python --version` (Linux/macOS) or `py --version` (Windows) to check your Python version reports 3.8.x.
+
+1. Install pip packages
+    ```sh
+    $ pip install -r requirements.txt
+    ```
+
+## Configure Azure Machine Learning Event 
+
+Once you completed prerequisites, configure the Azure Machine Learning Event.
+
+- [Deploy the function project to Azure](#Deploy-the-function-project-to-Azure)
+- [Configure Events](#Configure-Events)
+
+### Deploy the function project to Azure
+1. Install [Azure CLI 2.4 or later](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli). Make sure to check the version by running `az --version`.
+
+1. Login to Azure:
+    ```sh
+    $ az login
+    ```
+
+1. Check which subscription you are currently using:
+    ```sh
+    $ az account show -o table
+    ```
+
+    If you are not in the subscription you want to use, use `az account set -s [subscription id]` to switch.
+    You can also use ```az account list``` to get all the subscription id.
+
+1. Change directory to src directory.
+
+    ```sh
+    cd common/pipeline_monitor/src
+    ```
+
+1. Deploy your local functions project by using the `func azure functionapp publish` command. Replace <APP_NAME> with the name of your app.
+    ```sh
+    $ func azure functionapp publish <APP_NAME>
+    ```
+### Configure Events
+
+This function is triggered when Azure Machine Learning Events are issued. You can easily configure the event by following steps below:
+
+1. Open Azure Machine Learning resource from Azure Portal.
+
+1. Select **Events** menu.
+
+1. Click **+ Event Subscription**.
+
+1. Enter Name nad System Topic Name.
+
+1. Select **Run completed** and **Run status changed** from **EVENT TYEPS**
+
+1. Select **Azure Function** in **Eventpoint Type** and click **Select an endpoint**.
+
+1. Select created Azure Function app and **PipelineRunMonitor** function.
+
+1. Complete the wizard by creating it.
+
+# Analyze run result
+
+Once you completed deployment, you can analyze the run result.
+
+## Run Azure Machine Learning pipeline
+
+First of all, run any Azure Machine Learning pipeline and confirm the Fuction is triggered.
+
+## Query Application Insights
+
+It may take up to five minutes until the log is sent to Application Insights. 
+
+1. Open Application Insights resource.
+
+1. Select **Logs** in **Monitoring section**.
+
+1. Run Kusto query after replace variable and see the results.
```

Comment:
Similar comment here, maybe state in the text that the main query result is the pipeline run duration.

Reply:
@h2floh Just added additional explanations here as well.

Reply:
Thanks :)